testing/trino-tests/src/test/java/io/trino/tests/TestLocalDispatchQuery.java

```diff
@@ -0,0 +1,250 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import io.trino.Session;
+import io.trino.dispatcher.DispatchManager;
+import io.trino.dispatcher.DispatchQuery;
+import io.trino.dispatcher.LocalDispatchQuery;
+import io.trino.execution.QueryExecution;
+import io.trino.execution.QueryInfo;
+import io.trino.execution.QueryState;
+import io.trino.execution.StageId;
+import io.trino.execution.StateMachine;
+import io.trino.execution.TaskId;
+import io.trino.server.BasicQueryInfo;
+import io.trino.server.protocol.Slug;
+import io.trino.spi.QueryId;
+import io.trino.sql.planner.Plan;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.TestingSessionContext;
+import io.trino.tests.tpch.TpchQueryRunnerBuilder;
+import org.joda.time.DateTime;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.function.Consumer;
+
+import static io.trino.SessionTestUtils.TEST_SESSION;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestLocalDispatchQuery
+{
+    private static DispatchQuery localDispatchQuery;
+    private static CountDownLatch countDownLatch;
+
+    private DistributedQueryRunner queryRunner;
+
+    @BeforeClass
+    public void setUp()
+            throws Exception
+    {
+        queryRunner = TpchQueryRunnerBuilder.builder().build();
+        countDownLatch = new CountDownLatch(1);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        queryRunner.close();
+        queryRunner = null;
+    }
+
+    @Test(timeOut = 10_000L)
+    public void testDispatchedQuery()
+            throws Exception
+    {
+        DispatchManager dispatchManager = queryRunner.getCoordinator().getDispatchManager();
+        QueryId queryId = dispatchManager.createQueryId();
+        localDispatchQuery = (DispatchQuery) dispatchManager.createQuery(
+                queryId,
+                Slug.createNew(),
+                TestingSessionContext.fromSession(TEST_SESSION),
+                "SELECT * FROM lineitem",
+                Optional.of(new TestingQueryExecution()),
+                false)
+                .get();
+
+        // wait until query starts running
+        countDownLatch.await();
+        assertTrue(localDispatchQuery.getDispatchInfo().getCoordinatorLocation().isPresent());
```

Comment:
Does this test pass in this commit?  If not, please move merge it with the next commit, where the change is implemented

Reply:
I wanted to check if the test fails without the fix. Anyways, I have no objections to change the order of the commits. Will move it.